src/tests/Common/Directory.Build.targets

```diff
@@ -16,6 +16,7 @@
     <GCStressDependsOnCoreDisTools Condition="'$(TargetOS)' == 'Linux' And '$(TargetArchitecture)' == 'x64'">true</GCStressDependsOnCoreDisTools>
     <CopyCoreDisToolsToCoreRoot>false</CopyCoreDisToolsToCoreRoot>
     <CopyCoreDisToolsToCoreRoot Condition="$(GCStressDependsOnCoreDisTools) And '$(DotNetBuildFromSource)' != 'true'">true</CopyCoreDisToolsToCoreRoot>
+    <IsDesktopArchitecture Condition="'$(TargetArchitecture)' != 'wasm' and '$(TargetArchitecture)' != 'android' and '$(TargetArchitecture)' != 'iOS' ">true</IsDesktopArchitecture>
```

Comment:
Is this right?  Shouldn't it be comparing for `TargetOS` to `wasm`, `android`, `ios`, etc?

Also: shouldn't it check for `iossimulator` and `maccatalyst`, too?  ("Desktop" here means "doesn't use a custom host for testing", right?)

Reply:
Or better yet, `'$(TargetsiOS)' == 'true'` which means both `iOS` and `iOSSimulator`. Same the case with `tvOS`. Note that all non-libs subsets use these definitions https://github.com/dotnet/runtime/blob/c8ede5c4b1aba85fc894d5faed42d6d5186eabe1/Directory.Build.props#L190-L206 (and libs uses similar definitions by the SDK where we control SDK's internal hook in this repo: https://github.com/dotnet/runtime/blob/c8ede5c4b1aba85fc894d5faed42d6d5186eabe1/eng/targetframeworksuffix.props).